src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -222,6 +226,33 @@ public bool RecoverDatabase(TransactionHeader* txHeader, Action<string> addToIni
                 }
             }
 
+            if (_env.Options.EncryptionEnabled == false) // for encryption, we already use AEAD, so no need
+            {
+                // here we want to check that the checksum on all the modified pages is valid
+                // we can't do that during the journal application process because we may have modifications
+                // to pages that overwrite one another. So we have to do this at the end, this will detect
+                // corruption when applying journals at recovery time rather than at usage.
+                var tempTx = new TempPagerTransaction();
+
+                var sortedPages = modifiedPages.ToArray();
+                Array.Sort(sortedPages);
+
+                long minPageChecked = -1;
+
+                foreach (var modifedPage in modifiedPages)
```

Comment:
Usage of invalid variable in the loop - `modifiedPages` instead of `sortedPages`. That's why it wasn't working.

Already handled differently in https://github.com/ravendb/ravendb/pull/8303 and removed the overflow checks from here in https://github.com/ravendb/ravendb/pull/8308

Reply:
Another case: https://github.com/ravendb/ravendb/pull/8313